src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.netcoreapp.cs

```diff
@@ -24,5 +36,166 @@ public static void Create(this DirectoryInfo directoryInfo, DirectorySecurity di
+        /// <param name="options">One of the enumeration values that describes how to create or overwrite the file.</param>
+        /// <param name="fileSecurity">An object that determines the access control and audit security for the file.</param>
+        /// <returns>A file stream for the newly created file.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="mode" /> or <paramref name="share" /> are out of their legal enum range.
+        ///-or-
+        /// <paramref name="options" /> is not a supported enum value.
+        ///-or-
+        /// <paramref name="bufferSize" /> is not a positive number.</exception>
+        /// <exception cref="UnauthorizedAccessException">Access to the path is denied.</exception>
+        /// <exception cref="ArgumentException">Invalid handle.
+        ///-or-
+        /// Handle does not support asynchronous operations. The parameters to the <see cref="FileStream" /> constructor may need to be changed to indicate that the handle was opened synchronously (that is, it was not opened for overlapped I/O).
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">Cannot access a closed file.</exception>
+        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
+        {
+            if (fileInfo == null)
+            {
+                throw new ArgumentNullException(nameof(fileInfo));
+            }
+
+            if (fileSecurity == null)
+            {
+                throw new ArgumentNullException(nameof(fileSecurity));
+            }
+
+            // don't include inheritable in our bounds check for share
+            FileShare tempshare = share & ~FileShare.Inheritable;
+
+            string badArg = null;
+            if (mode < FileMode.CreateNew || mode > FileMode.Append)
+            {
+                badArg = nameof(mode);
+            }
+            else if (tempshare < FileShare.None || tempshare > (FileShare.ReadWrite | FileShare.Delete))
+            {
+                badArg = nameof(share);
+            }
+            if (badArg != null)
+            {
+                throw new ArgumentOutOfRangeException(badArg, SR.ArgumentOutOfRange_Enum);
+            }
+
+            // We need to check only the supported attributes
+            // NOTE: any change to FileOptions enum needs to be matched here in the error validation
+            if (options != FileOptions.None &&
+                (options & ~(FileOptions.WriteThrough | FileOptions.Asynchronous | FileOptions.RandomAccess | FileOptions.DeleteOnClose | FileOptions.SequentialScan | FileOptions.Encrypted | (FileOptions)0x20000000 /* NoBuffering */)) != 0)
```

Comment:
`NoBuffering` was commented in Framework, but if the user passes it combined, it was accepted.
See: https://referencesource.microsoft.com/#mscorlib/system/io/fileoptions.cs,35

The reason why we check for specific values is because when `options` is passed to the PInvoke as the `dwFlagsAndAttributes` parameter, there are many other values that are accepted but apparently not supported for files.
See: https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilew

@JeremyKuhne can confirm these explanations in case they are not accurate.

Reply:
There is a feature request to remove the checking completely. Personally I'd remove this block completely as we're blocking people more than we're helping them. If you've gone to the trouble to cast an int to this thing and call this advanced API, more power to you.

Reply:
Okay, I removed this check. I won't be checking that this enum is out of bounds.